server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -116,6 +119,7 @@ public PluginInfo(final StreamInput in) throws IOException {
             javaVersion = "1.8";
         }
         this.classname = in.readString();
+        this.folderName = in.readString();
```

Comment:
I think this needs to be scoped to a version (e.g. if a version is after 1.0.0) then read it from the input stream otherwise use the default value for bwc.

Also, you need to be add this field to the `writeTo` method for writing to the output stream .

Reply:
I missed that in my CR, good call.